ARCHITECTURE.md

```diff
@@ -139,6 +138,14 @@ corresponding `FooConnection`.
 Because `*Connection` classes are intended for customers to use (at least as
 mocks), they are part of the public API.
 
+There are two concrete versions of the (generally) abstract `*Connection` class:
```

Comment:
two? I see three listed.

Also, add: "These concrete implementations are not part of the public API." so readers know not to use them directly

Reply:
PTAL